src/Stratis.SmartContracts.Token/StandardToken.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Stratis.SmartContracts;
+using Stratis.SmartContracts.Token;
+
+public class StandardToken : SmartContract, IStandardToken
+{
+    public StandardToken(ISmartContractState smartContractState, uint totalSupply) 
+        : base(smartContractState)
+    {
+        this.TotalSupply = totalSupply;
+    }
+
+    public uint TotalSupply
+    {
+        get => PersistentState.GetUInt32(nameof(this.TotalSupply));
+        private set => PersistentState.SetUInt32(nameof(this.TotalSupply), value);
+    }
+
+    public uint GetBalance(Address address)
+    {
+        return PersistentState.GetUInt32($"Balance:{address}");
+    }
+
+    private void SetBalance(Address address, uint value)
+    {
+        PersistentState.SetUInt32($"Balance:{address}", value);
+    }
+
+    public bool Transfer(Address to, uint amount)
+    {
+        if (amount == 0)
+        {
+            Log(new TransferLog { From = Message.Sender, To = to, Amount = 0 });
+
+            return true;
+        }
+
+        uint senderBalance = GetBalance(Message.Sender);
+        
+        if (senderBalance < amount)
+        {
+            return false;
+        }
+
+        uint toBalance = GetBalance(to);
+
+        SetBalance(Message.Sender, senderBalance - amount);
+
+        checked
+        {
```

Comment:
Checked is enabled by default in our compiler - I wonder what we do to establish the best practice here: Suggest only using our compiler?

Reply:
I considered this, but figured it would be better practice for us to always use `checked` rather than rely on any specific compiler flags being set.